Assets/audio-and-voice-effects/AudioVoiceEffectsManager.cs

```diff
@@ -0,0 +1,162 @@
+    internal bool isChecked = false;
+
+    // Method to set up the Agora engine
+    public override void SetupAgoraEngine()
+    {
+        base.SetupAgoraEngine();
+
+        // Pre-load sound effects to improve performance
+        agoraEngine.PreloadEffect(soundEffectId, configData.soundEffectFilePath);
+
+        // Specify the audio scenario and audio profile
+        agoraEngine.SetAudioProfile(AUDIO_PROFILE_TYPE.AUDIO_PROFILE_DEFAULT, AUDIO_SCENARIO_TYPE.AUDIO_SCENARIO_CHATROOM);
+    }
+
+    // Method to start or stop audio mixing
+    public void StartAudioMixing(TMP_Text audioMixingBtnText)
+    {
+        audioPlaying = !audioPlaying;
+
+        if (audioPlaying)
+        {
+            audioMixingBtnText.text = "Stop Audio File";
+            agoraEngine.StartAudioMixing(configData.audioFilePath, false, -1);
+        }
+        else
+        {
+            agoraEngine.StopAudioMixing();
+            audioMixingBtnText.text = "Play Audio";
+        }
+    }
+
+    // Method to play or pause sound effects
+    public void PlaySoundEffect(TMP_Text playEffectBtnText)
+    {
+        if (soundEffectStatus == 0)
+        { // Stopped
+            agoraEngine.PlayEffect(
+                    soundEffectId,   // The ID of the sound effect file.
+                    configData.soundEffectFilePath,   // The path of the sound effect file.
+                    0,  // The number of sound effect loops. -1 means an infinite loop. 0 means once.
+                    1,   // The pitch of the audio effect. 1 represents the original pitch.
+                    0.0, // The spatial position of the audio effect. 0.0 represents that the audio effect plays in the front.
+                    100, // The volume of the audio effect. 100 represents the original volume.
+                    true,// Whether to publish the audio effect to remote users.
+                    0    // The playback starting position of the audio effect file in ms.
+            );
+            playEffectBtnText.text = "Pause Sound Effect";
+            soundEffectStatus = 1;
+        }
+        else if (soundEffectStatus == 1)
+        {   // Playing
+            agoraEngine.PauseEffect(soundEffectId);
+            soundEffectStatus = 2;
+            playEffectBtnText.text = "Resume Sound Effect";
+        }
+        else if (soundEffectStatus == 2)
+        {   // Paused
+            agoraEngine.ResumeEffect(soundEffectId);
+            soundEffectStatus = 1;
+            playEffectBtnText.text = "Pause Sound Effect";
+        }
+    }
+
+    // Method to apply voice effects
+    public void ApplyVoiceEffect(TMP_Text voiceEffectBtnText)
+    {
+        voiceEffectIndex++;
+        // Turn off all previous effects
+        agoraEngine.SetVoiceBeautifierPreset(VOICE_BEAUTIFIER_PRESET.VOICE_BEAUTIFIER_OFF);
+        agoraEngine.SetAudioEffectPreset(AUDIO_EFFECT_PRESET.AUDIO_EFFECT_OFF);
+        agoraEngine.SetVoiceConversionPreset(VOICE_CONVERSION_PRESET.VOICE_CONVERSION_OFF);
```

Comment:
In the manager class, please make separate methods for each type of effect. Manage the method to call from the UI layer.